src/iOS/Avalonia.iOS/AutomationPeerWrapper.cs

```diff
@@ -0,0 +1,270 @@
+        {
+            _view = view;
+            _peer = peer ?? ControlAutomationPeer.CreatePeerForElement(view.TopLevel);
+
+            _peer.PropertyChanged += PeerPropertyChanged;
+            _peer.ChildrenChanged += PeerChildrenChanged;
+
+            AccessibilityContainer = _view;
+            AccessibilityIdentifier = _peer.GetAutomationId();
+        }
+
+        private static void UpdateName(AutomationPeerWrapper self)
+        {
+            AutomationPeer peer = self;
+            self.AccessibilityLabel = peer.GetName();
+        }
+
+        private static void UpdateHelpText(AutomationPeerWrapper self)
+        {
+            AutomationPeer peer = self;
+            self.AccessibilityHint = peer.GetHelpText();
+        }
+
+        private static void UpdateBoundingRectangle(AutomationPeerWrapper self)
+        {
+            AutomationPeer peer = self;
+            Rect bounds = peer.GetBoundingRectangle();
+            PixelRect screenRect = new PixelRect(
+                self._view.TopLevel.PointToScreen(bounds.TopLeft),
+                self._view.TopLevel.PointToScreen(bounds.BottomRight)
+                );
+            self.AccessibilityFrame = new CGRect(
+                screenRect.X, screenRect.Y,
+                screenRect.Width, screenRect.Height
+                );
+        }
+
+        private static void UpdateIsReadOnly(AutomationPeerWrapper self)
+        {
+            AutomationPeer peer = self;
+            self.AccessibilityRespondsToUserInteraction =
+                peer.GetProvider<IValueProvider>()?.IsReadOnly ??
+                peer.GetProvider<IRangeValueProvider>()?.IsReadOnly ??
+                peer.IsEnabled();
+        }
+
+        private static void UpdateValue(AutomationPeerWrapper self)
+        {
+            AutomationPeer peer = self;
+            string? newValue =
+                peer.GetProvider<IRangeValueProvider>()?.Value.ToString("0.##") ??
+                peer.GetProvider<IValueProvider>()?.Value;
+            if (self.AccessibilityValue != newValue)
+            {
+                self.AccessibilityValue = newValue;
+                UIAccessibility.PostNotification(UIAccessibilityPostNotification.Announcement, self);
+            }
+        }
+
+        private void PeerChildrenChanged(object? sender, EventArgs e) => _view.UpdateChildren(_peer);
+
+        private void PeerPropertyChanged(object? sender, AutomationPropertyChangedEventArgs e) => UpdateProperties(e.Property);
+
+        private void UpdateProperties(params AutomationProperty[] properties)
+        {
+            Action<AutomationPeerWrapper>? setter =
+                Delegate.Combine(properties
+                .Where(s_propertySetters.ContainsKey)
+                .Select(x => s_propertySetters[x])
+                .Distinct()
+                .ToArray()) as Action<AutomationPeerWrapper>;
+            setter?.Invoke(this);
```

Comment:
This is expensive to compute on each property change.
It can be changed into a for-loop with HashSet to keep track of already invoked actions.

Reply:
Will do!